keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py

```diff
@@ -97,12 +98,14 @@ def __init__(
         segmentation_head=None,
         **kwargs,
     ):
-        if not isinstance(backbone, keras.layers.Layer) or not isinstance(
-            backbone, keras.Model
+        if (
```

Comment:
This logic seems incorrect. Won't this now fail if we use something that's not a `KerasCV` backbone since this is all `or`s?

I think we should just verify that the model is a `keras.layers.Layer`, which is true of all models and backbones.

I'm a bit confused as to why this failed locally for you in the first place. What version of KerasCV and KerasCore do you have installed?

Reply:
Hmm, my KCV is from the `master` branch and Keras Core is 0.1.2:

```
print(keras_cv.__version__) # 0.6.1
print(keras.__version__) # 0.1.2
```

Let me re-verify and come back again - this is probably a local issue then

Reply:
So I borked both packages and re-installed from scratch and it looks like the issue is gone - even though they're the same versions.
I love Python dependencies so much

Looks like a local issue - sorry for raising this. My bad 🤦‍♂️ 
Closing

Reply:
No worries!